src/shared/mount-util.c

```diff
@@ -742,3 +745,266 @@ int mount_option_mangle(
 
         return 0;
 }
+
+int bind_mount_in_namespace(
+                pid_t target,
+                const char *propagate_path,
+                const char *incoming_path,
+                const char *src,
+                const char *dest,
+                int read_only,
+                int make_file_or_directory,
+                char **error_message) {
```

Comment:
`error_message` is a weird thing, something we initialized on failure, which our typical return values are not. Hence I think it deserves a small comment here: `/* Initialized on failure *and* on success */`

or so. (there are probably other cases where we should document these special cases)

Reply:
added